### PR TITLE
Refactoring members naming in OperatorBase, derived classes and consumers

### PR DESCRIPTION
**Pull request by williamfgc**
## Proposed changes

Rename protected and private members of `OperatorBase`, derived classes and consumers
Member variables and functions to follow coding standards
Component of #3412
Related to #3411

## What type(s) of changes does this code introduce?

- Code style update (formatting, renaming)
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- Yes

## What systems has this change been tested on?
Ubuntu 20.04, must pass CI

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/Particle/DistanceTableData.h [hunk 1]
```diff
@@ -188,8 +188,8 @@ class DistanceTableData
    * @param P the target particle set
    */
   virtual void evaluate(ParticleSet& P) = 0;
-  virtual void mw_evaluate(const RefVectorWithLeader<DistanceTableData>& dt_list,
```
Comment (ye-luo) [inline src/Particle/DistanceTableData.h:191]: "mw_" prefix is a signature. `mw_evaluate` corresponds to `evaluate`. So only the piece after "mw_" should follow function naming convention.

**Review by ye-luo — CHANGES_REQUESTED** [review]
Renaming beyond OperatorBase. Please update title and description.

File: src/Particle/DistanceTableData.h (hunk 1 above)
Comment (williamfgc) [reply src/Particle/DistanceTableData.h:191]: @ye-luo  thanks, I'll revert these. Are there other special cases beside `mw_`? Asking so we can document these.

Comment (ye-luo) [reply src/Particle/DistanceTableData.h:191]: Also "flex_" is used for upper level batched APIs. I'm not aware of others.

File: src/QMCDrivers/WFOpt/HamiltonianRef.cpp [hunk 2]
```diff
@@ -59,14 +59,14 @@ int HamiltonianRef::addObservables(ParticleSet& P)
   P.Collectables.rewind();
   for (int i = 0; i < Hrefs_.size(); ++i)
     Hrefs_[i].get().addObservables(Observables, P.Collectables);
-  int myIndex = P.PropertyList.add(Observables.Names[0]);
+  int my_index_ = P.PropertyList.add(Observables.Names[0]);
```
Comment (ye-luo) [inline src/QMCDrivers/WFOpt/HamiltonianRef.cpp:62]: function local variable should not have underscore. Also add const if there is no intention to modify.

File: src/QMCHamiltonians/QMCHamiltonian.h [hunk 3]
```diff
@@ -97,7 +97,7 @@ class QMCHamiltonian
   void initialize_traces(TraceManager& tm, ParticleSet& P);
 
   // ///collect scalar trace data
-  //void collect_scalar_traces();
+  //void collectScalarTraces();
```
Comment (ye-luo) [inline src/QMCHamiltonians/QMCHamiltonian.h:100]: Simply remove.

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Pull request merged**